### DISCUSSION TITLE
Issues with adding the widget to the MIUI Home Screen

### DISCUSSION THREAD

**Discussion by Lijucay** [community/community#139993] [Mobile]
Has anyone gotten an issue with the MIUI Homescreen crashing after attempting to add the widget of GitHub to it? Cuz I certainly can't add it. Seems the problem is only on MIUI, not on other launchers.

![Screenshot_2024-09-28-00-06-05-944_com.miui.bugreport.jpg](https://github.com/user-attachments/assets/127e197f-b3c3-401f-836b-7b1fe8031401)

This is the error message I receive.

**Comment by brenoepics** [comment]
Same issue here

**Comment by Jackwillow786** [comment]
If you're having trouble adding a widget to the MIUI Home Screen, try restarting the phone or clearing the launcher cache. I had a similar issue once while setting up a [home repairs](https://pearllemonhomerepairs.co.uk/) app widget, and that worked!

**Reply by Lijucay** [reply]
This did not work for me unfortunately.

**Comment by JohnPullmann** [comment]
Same issue

**Comment by spacefren** [comment]
Same here, it's also happening in HyperOS

**Comment by IRNCyber** [comment]
Yes, there have been reports of issues with the GitHub widget crashing the MIUI homescreen. This seems to be a specific problem with MIUI and not other launchers.
The error message you've provided, "Screenshot_2024-09-28-00-06-05-944_com.miui.bugreport.jpg," indicates a crash within the MIUI system itself, likely triggered by the GitHub widget.
Here are some possible solutions:
 * Update MIUI: Ensure you're running the latest version of MIUI. Updates often include bug fixes and improvements.
 * Clear Cache and Data: Clearing the cache and data of the GitHub app and the MIUI Home app might resolve temporary glitches.
 * Disable and Re-enable the Widget: Try disabling the widget and then re-adding it. This can sometimes refresh the connection and fix issues.
 * Report the Issue: Report the bug to GitHub and Xiaomi. Providing detailed information about the issue, including the error message and device model, can help developers identify and fix the problem.
 * Use a Different Launcher: If the issue persists, consider switching to a different launcher like Nova Launcher or Microsoft Launcher. These launchers often have better compatibility and stability.
It's important to note that these are potential solutions, and the effectiveness may vary depending on the specific device and MIUI version. If the problem persists, reaching out to Xiaomi's support or the GitHub community for further assistance is recommended.

**Reply by brenoepics** [reply]
This answer seems AI-generated or overly generic and doesn’t address the actual problem. The crash is caused by a `NullPointerException` in the GitHub app due to improper handling of a nullable parameter in an `Intent`. This is unrelated to MIUI, as other apps widgets work fine. The issue lies in GitHub’s app code, not the launcher or OS.

The correct fix for @github-staff is probably to add proper null checks, as discussed in cases like [this StackOverflow thread](https://stackoverflow.com/questions/8373755/java-lang-runtimeexception-failure-delivering-result-resultinfowho-null-reque). Generic advice like updating MIUI or clearing caches won’t help here, as the root cause is in the app itself.

**Comment by jonatasperaza** [comment]
Still reproducible on a Xiaomi Redmi 12 running Android 15 as of July 19, 2026.

Steps to reproduce:

1. Open the Android home-screen widget picker.
2. Select the GitHub contributions widget.
3. Select the GitHub account for the widget.

Observed result: the Xiaomi System Launcher ("com.miui.home") crashes and restarts, and the widget is not added. The problem occurs consistently when attempting to add this specific GitHub widget.

There is also a newer report involving the same Redmi 12 device in discussion #182689.

Please investigate the GitHub widget’s compatibility with Xiaomi MIUI/HyperOS launchers and provide either a fix or a supported workaround.
